services/navigation/builtin/builtin.go

```diff
@@ -495,42 +509,86 @@ func (svc *builtIn) Close(ctx context.Context) error {
 	return svc.store.Close(ctx)
 }
 
+func (svc *builtIn) moveOnGlobeSync(ctx context.Context, wp navigation.Waypoint, extra map[string]interface{}) error {
+	req := motion.MoveOnGlobeReq{
+		ComponentName:      svc.base.Name(),
+		Destination:        wp.ToPoint(),
+		Heading:            math.NaN(),
+		MovementSensorName: svc.movementSensor.Name(),
+		Obstacles:          svc.obstacles,
+		MotionCfg:          svc.motionCfg,
+		Extra:              extra,
+	}
+	var stopWG sync.WaitGroup
+	defer stopWG.Wait()
+	cancelCtx, cancelFn := context.WithCancel(ctx)
+	defer cancelFn()
+	rawExecutionID, err := svc.motionService.MoveOnGlobeNew(cancelCtx, req)
+	if err != nil {
+		return err
+	}
+
+	executionID, err := uuid.Parse(rawExecutionID)
+	if err != nil {
+		return err
+	}
+	executionWaypoint := executionWaypoint{executionID: executionID, waypoint: wp}
+	if old := svc.activeExecutionWaypoint.Swap(executionWaypoint); old != nil && old != emptyExecutionWaypoint {
```

Comment:
This is a good idea.  Do we need to check this part though? `old != emptyExecutionWaypoint`

Reply:
It is unfortunately. https://pkg.go.dev/sync/atomic#Value.Swap
from the docs:
```
Swap stores new into Value and returns the previous value. It returns nil if the Value is empty.

All calls to Swap for a given Value must use values of the same concrete type. Swap of an inconsistent type panics, as does Swap(nil).
```

So the first time we call `Swap` we will get back `nil`, but after that point it will never return nil, b/c if I try to call `Swap(nil)` it panics. That is why we set it to `emptyExecutionWaypoint` after calling `StopPlan`.